### PR TITLE
Create GitHub CI workflow for arduino code

### PR DESCRIPTION
**Pull request by zmk5**
This code should attempt to download the `arduino-cli` onto an ubuntu 18.04 vm instance and compile the code to see if there are errors.

If this works properly, this CI can be used whenever a push or pull is made to the `master` or `develop` branches.

### CODE REVIEW

**Comment by zmk5** [comment]
It works now! 😃 This CI framework is rather simplistic but we can add to it easily for extra functionality. For now whenever a pull request is initiated or a push is made to either `develop` or `master`, this GH Action will attempt to compile the code for an Arduino Mega board with an `atmega2560` CPU. If an error is thrown up, we will know that something is wrong with the code.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - name: Update Arduino Core information
+      run: ./bin/arduino-cli core update-index
+    
+    # Install tools for the Arduino MEGA board
+    - name: Install Arduino Board tools
+      run: ./bin/arduino-cli core install arduino:avr
+    
+    # Compile the Arduino code and make sure it runs ok
+    - name: Compile Arduino code
+      run: ./bin/arduino-cli compile --fqbn arduino:avr:mega:cpu=atmega2560 software-controller.ino
```
Comment (mrrosen) [inline .github/workflows/main.yml]: Please add -v flag to make the compilation verbose; I fond it easier to debug if we can see the compilation steps :)

**Review by mrrosen — CHANGES_REQUESTED** [review]
Awesome, these kinds sanity checks will certain help our sanity :)

File: .github/workflows/main.yml (hunk 1 above)
Comment (zmk5) [reply .github/workflows/main.yml]: Will do!

**Review by mrrosen — APPROVED** [review]
Nice; I can now see the verbose compilation in the log

**Comment by zmk5** [comment]
Added the `-v` flag for compilation. If you go to the **Actions** tab at the top of this page and click the latest build you can see the verbose output for the code compilation.

**Pull request merged**